python/taichi/lang/kernel_impl.py

```diff
@@ -416,11 +416,24 @@ def extract_arg(arg, anno):
             shape = tuple(shape)
             element_shape = ()
             if isinstance(anno.dtype, MatrixType):
-                if len(shape) < anno.dtype.ndim:
-                    raise ValueError(
-                        f"Invalid argument into ti.types.ndarray() - required element_dim={anno.dtype.ndim}, "
-                        f"but the argument has only {len(shape)} dimensions")
+                if anno.ndim is not None:
```

Comment:
Can we add a few tests to test against these errors each? Thanks!

Reply:
Done!